src/rgw/rgw_admin.cc

```diff
@@ -8141,9 +8141,11 @@ int main(int argc, const char **argv)
     if (max_entries < 0)
       max_entries = 1000;
 
+    auto gen_id = bucket_info.layout.current_index.gen;
+
     do {
       list<rgw_bi_log_entry> entries;
-      ret = store->svc()->bilog_rados->log_list(bucket_info, shard_id, marker, max_entries - count, entries, &truncated);
+      ret = store->svc()->bilog_rados->log_list(bucket_info, shard_id, gen_id, marker, max_entries - count, entries, &truncated);
```

Comment:
we really need to differentiate between generation numbers for index layouts, and generation numbers for log layouts

here we're taking the current index generation and using it as a log generation. these are different things, even with `BucketLogType::InIndex`, because the 'bucket sync enable' command will be creating a new log generation that points to the same (older) index generation

so i think these interfaces should be explicit in taking either a log layout structure or an index layout structure so it's impossible to confuse the two

for example, this code would take the latest log generation `const auto& log = bucket_info.layout.logs.back()` and pass that to `log_list()`. when `log_list()` needs to call `open_bucket_index()`, it would then pass the index layout and gen stored under `log.layout.in_index`

Reply:
yes I had this question about passing the log gen. Thanks for clarifying!

Reply:
For RGWOp_BILog_List call to log_list(), I'm assuming that the log gen will be passed down through RGWSyncBucketShardCR() as part of Adam's datalog work.

Reply:
In particular, for `RGWOp_BILog_List `call to log_list(), I assumed that a log generation number will be passed down through Incremental sync as part of Adam's datalog work, in which case, bilog list would need to take a log gen number right? Then we would want to be able to get the corresponding index layout and gen values. I can get the values from logs.back() only if we are always defaulting to the latest generation.

Reply:
right! when we get a log generation number from an API or admin command, we can search `bucket_info.layout.logs` for a matching `bucket_log_layout_generation::gen` and use that

this comment was about using `auto gen_id = bucket_info.layout.current_index.gen;` directly, but github didn't show that part with my comment

for radosgw-admin, we should also add a `--gen` option for the `BILOG_LIST` command that defaults to the latest

Reply:
sure, understood

Reply:
changed to log generation. yet to handle the generation matching in open_bucket_index()